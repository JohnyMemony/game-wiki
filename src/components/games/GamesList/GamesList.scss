.games-list {
  &__items {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
  }

  &__pagination {
    display: flex;
    justify-content: flex-end;
  }
}
