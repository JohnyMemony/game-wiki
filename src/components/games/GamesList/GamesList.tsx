import React from 'react';
import { GameCard } from '../GameCard/GameCard';
import { Game } from '../../../models/games/Game';
import { Pagination } from '../../_shared/Pagination/Pagination';

import './GamesList.scss';

interface Props {
  items: Game[];
  page: number;
  total: number;
  onPageChange(page: number): void;
}

export const GamesList: React.FC<Props> = (props) => {
  const { items, page, total, onPageChange } = props;

  const renderItems = () => {
    return items.map((item, index) => {
      return <GameCard data={item} key={index} />;
    });
  };

  return (
    <div className="games-list">
      <div className="games-list__items">{renderItems()}</div>
      <div className="games-list__pagination">
        <Pagination page={page} total={total} onChange={onPageChange} />
      </div>
    </div>
  );
};
