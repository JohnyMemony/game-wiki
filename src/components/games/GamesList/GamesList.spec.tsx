import React from 'react';
import { shallow } from 'enzyme';
import { GamesList } from './GamesList';

describe('GamesList component', () => {
  it('should render GamesList component', function () {
    const component = shallow(<GamesList page={1} total={10} onPageChange={() => {}} items={[]} />);

    expect(component).toMatchSnapshot();
  });
});
