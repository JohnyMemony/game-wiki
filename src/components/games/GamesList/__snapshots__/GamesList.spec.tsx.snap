// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesList component should render GamesList component 1`] = `
<div
  className="games-list"
>
  <div
    className="games-list__items"
  />
  <div
    className="games-list__pagination"
  >
    <Pagination
      onChange={[Function]}
      page={1}
      total={10}
    />
  </div>
</div>
`;
